packages/odyssey-react-mui/src/labs/DataComponents/TableSettings.tsx

```diff
@@ -0,0 +1,138 @@
+
+export type TableSettingsProps = {
+  setTableState: Dispatch<SetStateAction<TableState>>;
+  tableOptions: TableProps;
+  tableState: TableState;
+};
+
+const TableSettings = ({
+  setTableState,
+  tableOptions,
+  tableState,
+}: TableSettingsProps) => {
+  const { t } = useTranslation();
+
+  const { hasChangeableDensity, hasColumnVisibility, columns } = tableOptions;
+  const { rowDensity, columnVisibility } = tableState;
+
+  const changeRowDensity = useCallback(
+    (value: (typeof densityValues)[number]) =>
+      (_event: React.MouseEvent<HTMLLIElement>) => {
+        // This is necessary to avoid linter errors, while the _event is necessary to satisfy the onClick type
+        if (process.env.NODE_ENV === "development") console.debug(_event);
+
+        setTableState((prevState) => ({
+          ...prevState,
+          rowDensity: value,
+        }));
+      },
+    [setTableState],
+  );
+
+  const changeColumnVisibility = useCallback(
+    (columnId: string) => (_event: React.MouseEvent<HTMLLIElement>) => {
+      // This is necessary to avoid linter errors, while the _event is necessary to satisfy the onClick type
+      if (process.env.NODE_ENV === "development") console.debug(_event);
+
+      setTableState((prevState) => ({
+        ...prevState,
+        columnVisibility: {
+          ...prevState.columnVisibility,
+          [columnId]: prevState.columnVisibility
+            ? prevState.columnVisibility[columnId] === false
+            : false,
+        },
+      }));
+    },
+    [setTableState],
+  );
+
+  const isColumnVisibilityChecked = useMemo(() => {
+    return columns.reduce(
+      (acc, column) => {
+        const isChecked = columnVisibility
+          ? columnVisibility[column.accessorKey!] !== false
+          : true;
+        acc[column.accessorKey!] = isChecked;
+        return acc;
+      },
+      {} as Record<string, boolean>,
+    );
+  }, [columns, columnVisibility]);
+
+  return (
+    <>
+      {hasChangeableDensity && (
+        <MenuButton
+          ariaLabel={t("table.density.arialabel")}
+          endIcon={<ListIcon />}
+          menuAlignment="right"
+          shouldCloseOnSelect={false}
+        >
+          <>
+            {densityValues.map((value: (typeof densityValues)[number]) => (
+              <MenuItem
+                key={value}
+                isSelected={rowDensity === value}
+                onClick={changeRowDensity(value)}
+              >
+                {`${value.charAt(0).toUpperCase()}${value.slice(1)}`}
+              </MenuItem>
+            ))}
+          </>
+        </MenuButton>
+      )}
+
+      {hasColumnVisibility && (
+        <MenuButton
+          ariaLabel={t("table.columnvisibility.arialabel")}
+          endIcon={<ShowIcon />}
+          menuAlignment="right"
+          shouldCloseOnSelect={false}
+        >
+          <>
+            {columns
+              .filter((column) => column.enableHiding !== false)
```

Comment:
Is this the same logic or am I missing something?
```suggestion
            {columns
              .filter((column) => Boolean(column.enableHiding))
```